### PR TITLE
Improved pubspec URI handling.

### PR DESCRIPTION
**Pull request by pq**

### CODE REVIEW

**Comment by pq** [comment]
@stereotype441 PTAL?

File: lib/src/linter.dart [hunk 1]
```diff
@@ -325,9 +331,11 @@ class SourceLinter implements DartLinter, AnalysisErrorListener {
 
   @override
   Iterable<AnalysisErrorInfo> lintPubspecSource(
-      {String contents, String sourceUrl}) {
+      {String contents, String sourcePath}) {
     var results = <AnalysisErrorInfo>[];
 
+    Uri sourceUrl = sourcePath == null ? null : p.toUri(sourcePath);
```
Comment (stereotype441) [inline lib/src/linter.dart]: Nit: maybe call it `sourceUri`?  It's confusing to have its type be `Uri` but to call it a "Url".

**Comment by stereotype441** [comment]
lgtm

File: lib/src/linter.dart (hunk 1 above)
Comment (pq) [inline lib/src/linter.dart]: `url` is consistent with the `yaml` API I eventually call and that was my thinking but I tend to agree!

**Pull request merged**